mvc/Controller/commands.php

```diff
@@ -59,4 +59,25 @@ function XcreateGallery()
         header("Location: ./?view=model");
     }
     return "Angaben sind nicht komplett";
-}
\ No newline at end of file
+}
+
+function XuploadImage(){
+    if (isset($_POST['title'], $_POST['tags'], $_POST['picture'], $_POST['galleryid']))
+    {
+        //$galleryId, $tag, $title, $pictureUnconverted, $thumbnailUnconverted
+        
+        $p = new Picture();
+    
+        $allowed_extensions = array('png', 'jpg', 'jpeg');
+        if(!in_array($_POST['picture'], $allowed_extensions))
```

Comment:
Im not sure but i think only the extention is needed.
I would recommend this:
`$ext = end((explode(".", $_POST['picture'])));
if(in_array($ext, $allowed_extension)){} `

By the way i found an mistake made by me: there shouldn't be a **!** in the if.
I will fix and finish it!